titus_isolate/event/kubernetes_opportunistic_window_publisher.py

```diff
@@ -36,6 +39,12 @@ def __init__(self):
         self.__custom_api = kubernetes.client.CustomObjectsApi(
             kubernetes.config.new_client_from_config(config_file=kubeconfig))
 
+        self.__lock = Lock()
+        self.__opportunistic_resources = {}
+
+        watch_thread = Thread(target=self.__watch)
+        watch_thread.start()
```

Comment:
I don't know how Python threading works, but is this the same as Java, where if one thread doesn't exit, it prevents the whole process from finishing?

Reply:
no.  We have several threads in `titus-isolate`.  Many of which are expected to never exit.  `titus-isolate` exits / restarts on command.